src/transformers/models/llama/convert_llama_weights_to_hf.py

```diff
@@ -102,7 +102,7 @@ def write_model(model_path, input_base_path, model_size, tokenizer_path=None, sa
     if base > 10000.0:
         max_position_embeddings = 16384
     else:
-        max_position_embeddings = 2048
+        max_position_embeddings = 4096
```

Comment:
A different logic condition should be used here. It should only default to this value in the llama2 case. We need to keep the previous condition in case the user is converting llama 1 weights. 

One option is adding a flag, so the user can specify if they're converting v1 or v2 weights

Reply:
Shall I add an argument in `write_model` which defaults to 'v1' or 'v2' then ?

Reply:
Yes, you should add a new argument to `write_model` and also an argument that can be passed in the CLI e.g. `--llama-version`, which can be 1 or 2. It should default to 1